### ISSUE TITLE
Fix user login for initial user

### ISSUE THREAD

**Issue by say-paul** [fedora-iot/fido-device-onboard-rs#321]
fix for #270 does not solve the issue initial user  created during the onboarding process not able to login as it asks for password which not set via the useradd command.

Probable solution will be to disable password and set login only via cert.

**Comment by runcom** [comment]
@say-paul can you describe what's happening? if I add a user + an ssh key I should be able to just login via the key - can you test this  out now that your PR has merged and report where we fail?

**Comment by nullr0ute** [comment]
If this is the bug I mentioned it's for local login and sudo as you can't use a key for local console based login (think a kiosk or similar) or for sudo so we need to be able to optionally also set a password.

**Comment by runcom** [comment]
> If this is the bug I mentioned it's for local login and sudo as you can't use a key for local console based login (think a kiosk or similar) or for sudo so we need to be able to optionally also set a password.

yes, I'm not questioning the feature itself or the use, it makes complete sense to support this - I wanted to check what is really going on by adding a user and ssh key and then trying to login over ssh

**Comment by runcom** [comment]
again no questioning there :) one issue is surely around the fact that with sayan's #270 I'm still not able to ssh into the system by using the key injected there with that user :) that is something I want to investigate, whether that's an osdefault or not it'll make my suspicion a bug or not

**Comment by say-paul** [comment]
[this](https://github.com/shadow-maint/shadow/issues/222) should be the answer for this issue, though I found out about the default pass today only, but I have tested that it indeed asks for password , which I have never set.

**Comment by say-paul** [comment]
`cat /etc/passwd | grep testuser` shows up as
`testuser:x:1001:1001::/var/home/testuser:/bin/bash` , it creates a entry in /etc/shadow as determined by :x:

**Comment by runcom** [comment]
What I’m asking is “with #270 can you login via ssh at least?”

**Comment by say-paul** [comment]
No, it asks for password instead.

**Comment by runcom** [comment]
I'd go and check on a live system built with IB and using simplified installer to validate this whole flow then

**Comment by say-paul** [comment]
Tested it with the main branch:

1. Ssh works without password.
2. Su - <testuser> asks for password.

**Comment by runcom** [comment]
good, as long as ssh works w/o a password, #270 works as expected and we could continue with this in the future

**Comment by runcom** [comment]
for real-world-like e2e test I would definitely rely on osbuild-composer upstream test where we fully setup the infra to run fdo too - although having something here w/o testing simplified installer is also needed
